File: driver-portal/src/ViewModels/Callback.cs
using static Rsbc.Dmf.CaseManagement.Service.Callback.Types;

namespace Rsbc.Dmf.DriverPortal.ViewModels;

public class Callback
{
    public Guid Id { get; set; }
    public DateTimeOffset RequestCallback { get; set; }

    public string Subject { get; set; }

    public string Description { get; set; }

    // TODO should not be using proto enum in view models
    public CallbackCallStatus CallStatus { get; set; }
    public DateTimeOffset Closed { get; set; }

    //public string Phone { get; set; }
    public Api.PreferredTime PreferredTime { get; set; }
}

File: shared-portal-ui/projects/core-ui/src/lib/api/models/callback.ts
/* tslint:disable */
/* eslint-disable */
import { CallStatus } from '../models/call-status';
import { PreferredTime } from '../models/preferred-time';
export interface Callback {
  callStatus?: CallStatus;
  closed?: string;
  id?: string;
  preferredTime?: PreferredTime;
  requestCallback?: string;
  subject?: string | null;
  description?: string | null;
}
